sympy/polys/polytools.py

```diff
@@ -2669,6 +2675,128 @@ def subresultants(f, g):
 
         return list(map(per, result))
 
+    def subresultant_polys(f, g):
+        """
+        Computes the subresultant polynomials of two polynomials ``f`` and ``g``.
+        The returned list has length min(deg(f), deg(g))+1, and is in order of
+        increasing degree. The first polynomial in the list is the resultant.
+
+        This uses the subresultant PRS. The remainder r_i is the deg(r_{i-1})-th
+        subresultant polynomial. Additionally, if deg(r_i) < deg(r_{i-1}) - 1,
+        then the deg(r_i) subresultant polynomial is r_i * LC(r_i) ^ c_i, where
+        c_i = deg(r_{i-1})-deg(r_i)-1.
+
+        Examples
+        ========
+
+        >>> from sympy import Poly
+        >>> from sympy.abc import x
+
+        >>> Poly(x**2 + 1, x).subresultant_polys(Poly(x**2 - 1, x))
+        [Poly(4, x, domain='ZZ'),
+         Poly(-2, x, domain='ZZ'),
+         Poly(x**2 - 1, x, domain='ZZ')]
+
+        Observe that the first element is the resultant.
+
+        >>> Poly(x**2 + 1, x).resultant(Poly(x**2 - 1, x))
+        4
+
+        See also
+        ========
+
+        subresultants
+        subresultant_coeffs
+
+        """
+
+        # ensure deg(f) >= deg(g)
+        n = f.degree()
+        m = g.degree()
+
+        if n < m:
+            f, g = g, f
+            n, m = m, n
+
+        prs = f.subresultants(g)
+        if len(prs) <= 1:
+            return []
+
+        subres_polys = [Poly(0, f.gen) for _ in range(m + 1)]
+
+        subres_polys[-1] = g * g.LC() ** (n - m - 1)
+
+        for i in range(2, len(prs)):
+            curr = prs[i]
+            prev = prs[i-1]
+            curr_deg = curr.degree()
+            prev_deg = prev.degree()
+
+            # remainder r_i is the deg(r_{i-1})-th subres poly
+            subres_polys[ prev_deg -1 ] = curr
+
+            # if there is a "degree jump"
+            # if deg(r_i) < deg(r_{i-1}) - 1, then the deg(r_i) subres poly
+            # is r_i * LC(r_i) ^ c_i, where c_i = deg(r_{i-1})-deg(r_i)-1
+            degree_jump = prev_deg - curr_deg - 1
+            if degree_jump > 0:
+                subres_polys[curr_deg] = curr * curr.LC() ** degree_jump
```

Comment:
Are there any test cases in which the degree jump is greater than 1?

There don't seem to be any and I am trying to work out exactly what the behaviour should be in that case.

Reply:
Here's an example:
```
In [16]: subresultants(x**5 + x**3 - 1, x**3)
Out[16]: [x**5 + x**3 - 1, x**3, 1]
```

At `i=2`, we have prev_deg-1 = 2, so the 2nd element of subres_polys is x^3. Furthermore, the degree jump is 3-0-1=2. So, the 0th element of subres_polys is 1 * 1 ^ 2 = 1. 

So that:
```
In [19]: subresultant_polys(x**5 + x**3 - 1, x**3)
Out[19]: [1, 0, 1, x**3]
```

In this case it seems inconsequential because its just 1. But it becomes important in the multivariate setting.
```
In [18]: subresultant_polys(x**5 + x**3 - 1, y*x**3)
Out[18]: [y**3, 0, y**3, x**3*y]
```
Where the 0th element is now y * y^2 = y^3.

Reply:
Here's an example:
```
In [19]: subresultants(x**5 + x**3 - 1, 2*x**3)
Out[19]: [x**5 + x**3 - 1, 2*x**3, 8]
```

At i=2, the degree jump deg(r_{i-1}) - deg(r_i) - 1 = 2. I've now added this as a test case. But, in this case, the 0th element was not matching the Mathematica output for the same (it was 512 I think, vs 32 which is correct, so basically off by some exponents). I spent a lot of time digging around on what the right form of this formula is, but I couldn't figure it out. I'm sure I'm missing something minor, but I'm not sure what. But eventually, I just used the fact that the 0th element is always the resultant. It's now giving correct output but I'm not sure if this is fully right.